src/main/java/org/assertj/core/extention/SoftlyExtension.java

```diff
@@ -0,0 +1,112 @@
+import org.junit.platform.commons.support.AnnotationSupport;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+
+/**
+ * <p>
+ * Injects an instance of {@link SoftAssertions} to any Field of type
+ * {@code SoftAssertions} that has the annotation {@link Softly}
+ * before each test method execution and invokes
+ * {@link SoftAssertions#assertAll()} after the test method
+ * </p>
+ * <p>
+ * This extension will throw a {@link Failures#failure(String)}
+ * when more that one filed is annotated with{@code @Softly}.
+ * </p>
+ * <p>
+ * A nested test class can provide a {@code SoftAssertions}
+ * field when it extends {@code this} extension or can inherit
+ * the parent's {@code Soft assertions field}</p>
+ * <p>
+ * Inspired by Mockito @Captor annotation.
+ * </p>
+ *
+ * @author Arthur Mita
+ **/
+public class SoftlyExtension implements TestInstancePostProcessor, BeforeEachCallback, AfterTestExecutionCallback {
+
+  private static final Logger LOGGER = Logger.getLogger(SoftlyExtension.class.getName());
+  private static final Namespace ASSERTJ_CORE = Namespace.create("org.assertj.core");
+  private static final String TEST_INSTANCE = "testInstance";
+  private final Set<SoftAssertions> softlies = new HashSet<>();
+
+  private Failures failures = Failures.instance();
+
+  @Override
+  public void postProcessTestInstance(Object testInstance, ExtensionContext extensionContext) throws Exception {
+    extensionContext.getStore(ASSERTJ_CORE).put(TEST_INSTANCE, testInstance);
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) throws Exception {
+    Optional<ExtensionContext> currentContext = Optional.of(extensionContext);
+
+    while (currentContext.isPresent()) {
+      ExtensionContext context = currentContext.get();
+      Object testInstance = context.getStore(ASSERTJ_CORE).remove(TEST_INSTANCE);
+
+      if (testInstance != null) {
+
+        List<Field> softlyFields = AnnotationSupport.findAnnotatedFields(testInstance.getClass(), Softly.class);
+
+        if (!softlyFields.isEmpty()) {
+
+          if (softlyFields.size() > 1) {
+            throw failures.failure("Only one field of type " + SoftAssertions.class.getName() + " can have the annotation @Softly");
+          }
+
+          Field softlyField = softlyFields.get(0);
+          if (softlyField.getType() != SoftAssertions.class) {
+            throw failures.failure("@Softly field must be of the type " + SoftAssertions.class.getName());
+          }
+          softlyField.setAccessible(true);
+          softlyField.set(testInstance, new SoftAssertions());
+
+          this.softlies.add((SoftAssertions) softlyField.get(testInstance));
+        }
+      }
+      currentContext = context.getParent();
```

Comment:
is that for nested tests?

Reply:
Yes. Nested test classes that have parents whose test instance lifecycle is not `LifeCycle.PER_CLASS` cannot use the SoftAssertions field declared in the parent cause they do not share the same instance. One would get an `IllegalArgumentException`